relay-server/src/actors/store.rs

```diff
@@ -224,27 +83,69 @@ impl Producers {
-                config,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            events: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::Events)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
                 &mut reused_producers,
-                KafkaTopic::MetricsTransactions,
-            )?,
-            profiles: make_producer(config, &mut reused_producers, KafkaTopic::Profiles)?,
-            replay_recordings: make_producer(
-                config,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            transactions: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::Transactions)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
                 &mut reused_producers,
-                KafkaTopic::ReplayRecordings,
-            )?,
-            replay_events: make_producer(config, &mut reused_producers, KafkaTopic::ReplayEvents)?,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            sessions: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::Sessions)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            metrics_sessions: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::MetricsSessions)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            metrics_transactions: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::MetricsTransactions)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            profiles: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::Profiles)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            replay_recordings: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::ReplayRecordings)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
+            replay_events: Producer::create(
+                &config
+                    .kafka_config(KafkaTopic::ReplayEvents)
+                    .map_err(|_| ServerErrorKind::KafkaError)?,
+                &mut reused_producers,
+            )
+            .map_err(|_| ServerErrorKind::KafkaError)?,
```

Comment:
Another nit: These expressions all follow the same pattern, should we replace them with a helper function that just takes a `KafkaTopic` as argument?